lib/server_configfile.ml

```diff
@@ -1,19 +1,15 @@
 open Lwt.Syntax
 
-type build_with =
-  | Opam
-  | Dune
-
 let yaml_of_build_with = function
-  | Opam -> `String "opam"
-  | Dune -> `String "dune"
+  | Intf.Build_with.Opam -> `String "opam"
+  | Intf.Build_with.Dune -> `String "dune"
```

Comment:
Should this function be extracted to `Intf`?

Reply:
`Intf` doesn't have any YAML conversion code so I don't think that would be the right place to live (unless we move everything there).